sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -994,6 +994,88 @@ class PlannerSuite extends SharedSparkSession with AdaptiveSparkPlanHelper {
       }
     }
   }
+
+  test("EnsureRequirements.reorder should fallback to the right side HashPartitioning") {
+    val plan1 = DummySparkPlan(
+      outputPartitioning = HashPartitioning(exprA :: exprB :: exprC :: Nil, 5))
+    val plan2 = DummySparkPlan(
+      outputPartitioning = HashPartitioning(exprB :: exprC :: Nil, 5))
+    // The left keys cannot be reordered to match the left partitioning, and it should
+    // fall back to reorder the right side.
```

Comment:
Could you add tests for the case where `reorder` fallbacks from right `HashPartitioning ` to `PartitioningCollection `. I think its better to cover all the fallback scenarios as much as possible.

Reply:
Added. Thanks!